pkg/controller/queuejob/queuejob_controller_ex.go

```diff
@@ -1274,26 +1285,30 @@ func (qjm *XController) ScheduleNext() {
 					} else {
 						fits = false
 						//Quota manager not initialized
-						dispatchFailedMessage = "Quota evaluation is enable but not initialize. Insufficient quota to dispatch AppWrapper."
-						klog.Errorf("[ScheduleNext] Quota evaluation is enable but not initialize.  AppWrapper %s/%s does not have enough quota\n", qj.Name, qj.Namespace)
+						dispatchFailedMessage = "Quota evaluation is enabled but not initialized. Insufficient quota to dispatch AppWrapper."
+						klog.Errorf("[ScheduleNext] Quota evaluation is enabled but not initialized.  AppWrapper %s/%s does not have enough quota\n", qj.Name, qj.Namespace)
 					}
 				} else {
-					klog.V(10).Infof("[ScheduleNext] HOL quota evaluation not enabled for %s at %s activeQ=%t Unsched=%t &qj=%p Version=%s Status=%+v due to quota limits", qj.Name, time.Now().Sub(HOLStartTime), qjm.qjqueue.IfExistActiveQ(qj), qjm.qjqueue.IfExistUnschedulableQ(qj), qj, qj.ResourceVersion, qj.Status)
+					klog.V(4).Infof("[ScheduleNext] HOL quota evaluation not enabled for %s at %s activeQ=%t Unsched=%t &qj=%p Version=%s Status=%+v", qj.Name, time.Now().Sub(HOLStartTime), qjm.qjqueue.IfExistActiveQ(qj), qjm.qjqueue.IfExistUnschedulableQ(qj), qj, qj.ResourceVersion, qj.Status)
 				}
 
-				// If quota evalauation is set or quota evaluation not enabled set the appwrapper to be dispatched
+				// If quota evalauation sucedeed or quota evaluation not enabled set the appwrapper to be dispatched
 				if fits {
 					// aw is ready to go!
 					apiQueueJob, e := qjm.queueJobLister.AppWrappers(qj.Namespace).Get(qj.Name)
 					// apiQueueJob's ControllerFirstTimestamp is only microsecond level instead of nanosecond level
 					if e != nil {
 						klog.Errorf("[ScheduleNext] Unable to get AW %s from API cache &aw=%p Version=%s Status=%+v err=%#v", qj.Name, qj, qj.ResourceVersion, qj.Status, err)
+						if qjm.quotaManager != nil && quotaFits {
```

Comment:
@asm582, @dmatch01, @KPostOffice and @metalcycling I believe that this one of the causes for `client already allocated` set of errors.

cc: @tardieu

Reply:
quota is not released when AW is deleted by clients, which is a different issue.